src/pydicom/pixels/utils.py

```diff
@@ -2098,3 +2170,246 @@ def unpack_bits(src: bytes, as_array: bool = True) -> "np.ndarray | bytes":
         raise ValueError("unpack_bits() requires NumPy if 'as_array = True'")
 
     return b"".join(map(_UNPACK_LUT.__getitem__, src))
+
+
+def get_packed_frame(
+    src: bytes,
+    index: int,
+    frame_length: int,
+    pad: bool = True,
+) -> bytes:
+    """Retrieve a single bit-packed frame from a bit-packed buffer.
+
+    This should only be used for single bit data (Bits Allocated = 1) in a
+    "bit-packed" representation (one byte containing 8 pixels). The pixel data
+    of such images may have pixel data from multiple frames within a single
+    byte.
+
+    Parameters
+    ----------
+    src : bytes
+        The native bit-packed representation of a full PixelData element,
+        potentially including multiple frames.
+    index : int
+        Zero-based index of the frame to extract.
+    frame_length : int
+        Number of pixels in each frame.
+    pad : bool
+        Whether to zero-pad the resulting bytes to a multiple of 2 bytes.
+
+    Returns
+    -------
+    bytes | bytearray
+        Bit-packed representation of the single frame requested.
+
+    Note
+    ----
+    In certain situations, this function will run faster and use less memory if
+    NumPy is installed. Otherwise, it will fall back to a slower method using
+    native Python functions.
+    """
```

Comment:
Maybe check that `src` is sufficiently long enough to contain the frame being sought.

Reply:
Addressed in [e9ea3de](https://github.com/pydicom/pydicom/pull/2216/commits/e9ea3deb538f6786f5a80e0e826d7aaf7a40678c)